src/backend/allocator/gbm.rs

```diff
@@ -11,10 +11,108 @@ use super::{
 #[cfg(feature = "backend_drm")]
 use crate::backend::drm::DrmNode;
 use crate::utils::{Buffer as BufferCoords, Size};
-pub use gbm::{BufferObject as GbmBuffer, BufferObjectFlags as GbmBufferFlags, Device as GbmDevice};
+#[cfg(feature = "backend_drm")]
+use drm::buffer::PlanarBuffer;
+use gbm::BufferObject;
+pub use gbm::{BufferObjectFlags as GbmBufferFlags, Device as GbmDevice};
 use std::os::unix::io::{AsFd, BorrowedFd};
 use tracing::instrument;
 
+/// A GBM buffer object
+#[derive(Debug)]
+pub struct GbmBuffer {
+    bo: BufferObject<()>,
+    size: Size<i32, BufferCoords>,
+    format: Format,
+}
```

Comment:
This is a great commit, because I just ran into an issue with virgl, that wlroots works around by overriding the modifier: https://gitlab.freedesktop.org/wlroots/wlroots/-/commit/bf86110fc5615bfd1af46e95cf81ac720ecac307

This is a great place to do the same, if we end up calling `gbm_bo_create`.